qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -150,27 +152,39 @@ class AddIntoPhaseGrad(GateWithRegisters, cirq.ArithmeticGate):
         [Compilation of Fault-Tolerant Quantum Heuristics for Combinatorial Optimization](https://arxiv.org/abs/2007.07391)
         Appendix A: Addition for controlled rotations
     """
+
     x_bitsize: int
     phase_bitsize: int
     right_shift: int = 0
     sign: int = +1
+    controlled: bool = False
```

Comment:
We've used `controlled: Optional[bool] = None` in the past to denote 3 cases
a) None -  no controls
b) 1 - Controlled with value 1
c) 0 - Controlled with value 0